tests/integration/functional/zen_stores/test_zen_store.py

```diff
@@ -0,0 +1,991 @@
+        client.zen_store.update_stack(
+            stack_id=default_stack.id, stack_update=stack_update
+        )
+
+
+def test_deleting_default_stack_fails():
+    """Tests that deleting the default stack is prohibited."""
+    client = Client()
+
+    default_stack = client.get_stack(DEFAULT_STACK_NAME)
+    with pytest.raises(IllegalOperationError):
+        client.zen_store.delete_stack(default_stack.id)
+
+
+def test_get_stack_fails_with_nonexistent_stack_id():
+    """Tests getting stack fails with nonexistent stack id."""
+    client = Client()
+    store = client.zen_store
+
+    with pytest.raises(KeyError):
+        store.get_stack(uuid.uuid4())
+
+
+def test_filter_stack_succeeds():
+    """Tests getting stack."""
+    client = Client()
+    store = client.zen_store
+
+    with ComponentContext(
+        c_type=StackComponentType.ORCHESTRATOR, flavor="local", config={}
+    ) as orchestrator:
+        with ComponentContext(
+            c_type=StackComponentType.ARTIFACT_STORE, flavor="local", config={}
+        ) as artifact_store:
+            components = {
+                StackComponentType.ORCHESTRATOR: [orchestrator.id],
+                StackComponentType.ARTIFACT_STORE: [artifact_store.id],
+            }
+            with StackContext(components=components) as stack:
+                returned_stacks = store.list_stacks(
+                    StackFilterModel(name=stack.name)
+                )
+                assert returned_stacks
+
+
+def test_crud_on_stack_succeeds():
+    """Tests getting stack."""
+    client = Client()
+    store = client.zen_store
+
+    with ComponentContext(
+        c_type=StackComponentType.ORCHESTRATOR, flavor="local", config={}
+    ) as orchestrator:
+        with ComponentContext(
+            c_type=StackComponentType.ARTIFACT_STORE, flavor="local", config={}
+        ) as artifact_store:
+            components = {
+                StackComponentType.ORCHESTRATOR: [orchestrator.id],
+                StackComponentType.ARTIFACT_STORE: [artifact_store.id],
+            }
+            new_stack = StackRequestModel(
+                name="arias_stack",
+                components=components,
+                workspace=client.active_workspace.id,
+                user=client.active_user.id,
+            )
+            created_stack = store.create_stack(stack=new_stack)
+
+            stacks = store.list_stacks(StackFilterModel(name="arias_stack"))
+            assert len(stacks) == 1
+
+            with does_not_raise():
+                stack = store.get_stack(created_stack.id)
+                assert stack is not None
+
+            # Update
+            stack_update = StackUpdateModel(name="axls_stack")
+            store.update_stack(stack_id=stack.id, stack_update=stack_update)
+
+            stacks = store.list_stacks(StackFilterModel(name="axls_stack"))
+            assert len(stacks) == 1
+            stacks = store.list_stacks(StackFilterModel(name="arias_stack"))
```

Comment:
```suggestion
            stack_name = sample_name("arias_stack")
            new_stack = StackRequestModel(
                name=stack_name,
                components=components,
                workspace=client.active_workspace.id,
                user=client.active_user.id,
            )
            created_stack = store.create_stack(stack=new_stack)

            stacks = store.list_stacks(StackFilterModel(name=stack_name))
            assert len(stacks) == 1

            with does_not_raise():
                stack = store.get_stack(created_stack.id)
                assert stack is not None

            # Update
            stack_update = StackUpdateModel(name="axls_stack")
            store.update_stack(stack_id=stack.id, stack_update=stack_update)

            stacks = store.list_stacks(StackFilterModel(name="axls_stack"))
            assert len(stacks) == 1
            stacks = store.list_stacks(StackFilterModel(name=stack_name))
```

Should at least make the test parallelizable.